zone-detector-service/main.py

```
@@ -0,0 +1,291 @@
+    return {"status": "ok", "model": "loaded" if MODEL is not None else "failed"}
+
+
+@app.post("/detect")
+def detect(req: DetectRequest):
+    if MODEL is None:
+        raise HTTPException(status_code=503, detail="Zone-detector model not initialised")
+    try:
+        return _run_detection(req.pdfPath, req.jobId)
+    except FileNotFoundError as e:
+        raise HTTPException(status_code=422, detail=str(e))
+    except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)